core/src/main/java/apoc/cypher/CypherInitializer.java

```diff
@@ -46,23 +52,31 @@ public void available() {
         Util.newDaemonThread(() -> {
 
             try {
-                awaitApocProceduresRegistered();
+                final boolean isSystemDatabase = db.databaseName().equals(GraphDatabaseSettings.SYSTEM_DATABASE_NAME);
+                if (!isSystemDatabase) {
+                    awaitApocProceduresRegistered();
+                }
                 Configuration config = dependencyResolver.resolveDependency(ApocConfig.class).getConfig();
 
-                TreeMap<String, String> initializers = Iterators.stream(config.getKeys(ApocConfig.APOC_CONFIG_INITIALIZER_CYPHER))
-                        .collect(Collectors.toMap(k -> k, k -> config.getString(k),
-                                (v1, v2) -> {
-                                    throw new RuntimeException(String.format("Duplicate key for values %s and %s", v1, v2));
-                                },
-                                TreeMap::new));
+                TreeMap<String, String> initializers = new TreeMap<>();
+
+                config.getKeys(ApocConfig.APOC_CONFIG_INITIALIZER + "." + db.databaseName()).forEachRemaining(key -> initializers.put(key, config.getString(key)));
+
+                if (!isSystemDatabase) {
```

Comment:
can we move this higher as a guard clause and just return from the method?

also can this whole collecting initializers go into a separate method?

Reply:
moving higher doesn't make a lot of sense IMO. We do want to execute initializers for any db (including system). Systemdb is just treated differently.
Collecting initializers is now in a separate method, see my commit below.